golos.publication/golos.publication.cpp

```diff
@@ -264,10 +265,10 @@ void publication::update_message(structures::mssgid message_id,
 auto publication::get_pool(tables::reward_pools& pools, uint64_t time) {
     eosio_assert(pools.begin() != pools.end(), "publication::get_pool: [pools] is empty");
 
-    auto pool = pools.upper_bound(time);
+    auto pool = pools.find(time);
 
-    eosio_assert(pool != pools.begin(), "publication::get_pool: can't find an appropriate pool");
-    return (--pool);
+    eosio_assert(pool != pools.end(), "publication::get_pool: can't find an appropriate pool");
+    return pool;
```

Comment:
In this place will be return a copy of iterator, which makes `chaindb_clone()`
Request to change function to:
```
auto& publication::get_pool(tables::reward_pools& pools, uint64_t time) {
....
    return *pool;
}
```